lib/tic_tac_toe/web/game_session_plug.ex

```diff
@@ -0,0 +1,22 @@
+defmodule TicTacToe.Web.GameSessionPlug do
+  use Plug.Builder
+
+
+  def create_or_find_game(conn, _opts) do
+    maybe_game_state = conn |> fetch_session() |> get_session(:game_state)
```

Comment:
Not sure about the `maybe_` prefix here.

You could just have a method that asks if there's a game state. I would also hide these `fetch_session` `get_session` calls. It makes things more readable when your public functions have only the high level logic in them, and the low level details are pushed into private methods. 

```
if has_game_state? do
    conn
 else
   conn |> create_game_state!
end